Update README.md to scope more on ai-lab-helm-charts (#32)
redhat-ai-dev/ai-lab-helm-charts: README.md

# AI Lab Helm Charts

This project is a collection of helm charts, insipired by the [AI Software Templates](https://github.com/redhat-ai-dev/ai-lab-template).

## Gitops

The gitops component, handled by ArgoCD for the RHDH case, is replaced by a Kubernetes Job created by the Helm chart, where this Job:

- Creates the GitHub repository for the application.
- Copies the application source code into the new repository.

> - Copies the Tekton `PipelineRun` that builds new images for the application as pull requests

Reviewer: ```suggestion
- Copies the Tekton `PipelineRun` that builds new images for the application after pull requests merge or commits are pushed directly to the `main` branch.
```

Per https://github.com/redhat-ai-dev/ai-lab-helm-charts/blob/main/pac/pipelineRuns/.tekton/docker-push.yaml#L6-L7 we don't set up to run PipelineRuns on PR creation.

Reviewer: by comparison, the pull request form of PAC:  https://github.com/redhat-ai-dev/ai-lab-template/blob/main/skeleton/source-repo/.tekton/docker-pull-request.yaml#L6-L7

> - Copies the Tekton Pipelines As Code PipelineRun/Pipeline/Task that build new images for the application as pull requests

Reviewer: Actually we just copy the PipelineRun.  All Pipelines and Task are remote ref's to the rhdh-pipelines repo

Author: Addressed in [e7dfb81](https://github.com/redhat-ai-dev/ai-lab-helm-charts/pull/32/commits/e7dfb814a567efc6592b44b65aa368868f8a7adb)

> - Copies the Tekton `PipelineRun` that builds new images for the application after pull requests merge or commits are pushed directly to the `main` branch.

Reviewer: ```suggestion
- Copies the Tekton `PipelineRun` that builds new images for the application after pull requests merge or commits are pushed directly to the `main` branch, 
```

>   are merged and updates the Deployment of the application with the new version of the image.

Reviewer: ```suggestion
  are merged and updates the Deployment of the application with the new version of the image by the PipelienRun directly patching the Deployment via `oc`, vs. ArgoCD using gitops to patch the Deployment.
```

>   are merged and updates the Deployment of the application with the new version of the image by the PipelienRun directly patching the Deployment via `oc`, vs. ArgoCD using gitops to patch the Deployment.

Reviewer: ```suggestion
and then updates the Deployment of the application with the new version of the image by directly patching the Deployment via `oc`, vs. ArgoCD using gitops to patch the Deployment.
```

  - The Pipelines As Code component is set up with the requisite credentials to process GitHub events for you GitHub Application.
- The [Pipeline Setup Helm Chart](/charts/ai-software-templates/pipeline-setup) that makes the User/Tenant changes in your application Namespace:
  - The Pipelines As Code PipelinesRuns in your Namespace have the requisite credentials to push your application's image to your Quay repository.
  - The [Gitops component](#gitops) has the requisite credentials to interact with the GitHub repository for your application.

> The ai-lab helm charts mainly use the tekton pipelines under [rhdh-pipelines](https://github.com/redhat-ai-dev/rhdh-pipelines) repo. The only customized resource used for the helm chart case is:

Reviewer: ```suggestion
The ai-lab helm charts exclusively uses tekton pipelines and tasks under [rhdh-pipelines](https://github.com/redhat-ai-dev/rhdh-pipelines) repo. The only customized resource used for the helm chart case is:
```
